File: dotnet-desktop-guide/wpf/app-development/snippets/how-to-use-host-builder/vb/SampleLifecycleService.vb
Imports System.Threading
Imports Microsoft.Extensions.Hosting

' <SampleLifecycleService>
Public Class SampleLifecycleService
    Implements IHostedService

    Public Function StartAsync(cancellationToken As CancellationToken) As Task Implements IHostedService.StartAsync
        System.Diagnostics.Debug.WriteLine("SampleLifecycleService: Started.")
        Return Task.CompletedTask
    End Function

    Public Function StopAsync(cancellationToken As CancellationToken) As Task Implements IHostedService.StopAsync
        System.Diagnostics.Debug.WriteLine("SampleLifecycleService: Stopped.")
        Return Task.CompletedTask
    End Function

End Class
' </SampleLifecycleService>

File: dotnet-desktop-guide/wpf/app-development/snippets/how-to-use-host-builder/csharp/SampleLifecycleService.cs
using Microsoft.Extensions.Hosting;

namespace HostBuilderApp;

// <SampleLifecycleService>
public class SampleLifecycleService : IHostedService
{
    public Task StartAsync(CancellationToken cancellationToken)
    {
        System.Diagnostics.Debug.WriteLine("SampleLifecycleService: Started.");
        return Task.CompletedTask;
    }

    public Task StopAsync(CancellationToken cancellationToken)
    {
        System.Diagnostics.Debug.WriteLine("SampleLifecycleService: Stopped.");
        return Task.CompletedTask;
    }
}
// </SampleLifecycleService>
